extensions/_states/heroku.py

```diff
@@ -0,0 +1,181 @@
+# -*- coding: utf-8 -*-
+'''
+State for managing heroku apps.
+
+.. versionadded:: 2015.5.0
+
+:configuration: This state can be used by either passing an api key directly
+    or by specifying it in a configuration profile in pillar.
+
+    It is possible to use a different API than http://api.heroku.com,
+    by specifying the API URL in config as api_url, or by passing the value directly.
+
+    For example:
+
+    .. code-block:: yaml
+
+        heroku.<name_of_state>:
+          - api_key: peWcBiMOS9HrZG15peWcBiMOS9HrZG15
+'''
+from __future__ import absolute_import
+import logging
+
+import sys
+import salt.config
+import salt.exceptions
+import salt.ext.six.moves.http_client
+import salt.syspaths
+import salt.utils
+
+log = logging.getLogger(__name__)
+
+# Global keys to maintain state of dict mismatch between Heroku and Pillar
+diff_heroku_pillar_dict = {'key_or_value_mismatch': False,
+                           'heroku_not_pillar': False,
+                           'pillar_not_heroku': False}
+
+# Global dicts to store diffs between Heroku and Pillar config vars
+heroku_diff_dict = {}
+pillar_diff_dict = {}
+
+
+def __virtual__():
+    '''
+    Return virtual name of the module.
+
+    :return: The virtual name of the module.
+    '''
+    return True
+
+
+def diff_app_config_vars(name, config_vars, api_key):
```

Comment:
This seems like a function that is a better fit for the execution module or a utils module.

Reply:
or at least prefixing it with `_`